internal/circonus/alerting.go

```diff
@@ -17,414 +17,370 @@ import (
-				logger.Warn().Err(err).Str("rule_id", rid).Interface("rule_cfg", ruleTemplate).Msg("creating rule")
-				continue // if there's an error creating, leave the old rule in place
-			}
-			logger.Debug().Str("rule_id", rid).Str("cid", origRule.CID).Msg("deleting old rule")
-			if _, err := client.DeleteRuleSetByCID(apiclient.CIDType(&origRule.CID)); err != nil {
-				logger.Warn().Err(err).Str("rule_id", rid).Interface("rule_cfg", origRule).Msg("deleting old rule")
-			}
-			continue
-		}
-
-		origData, err := json.Marshal(origRule)
-		if err != nil {
-			logger.Warn().Err(err).Interface("orig_rule", origRule).Msg("encoding original rule")
-			continue
-		}
-		mergedData, err := json.Marshal(ruleTemplate)
-		if err != nil {
-			logger.Warn().Err(err).Interface("merged_rule", ruleTemplate).Msg("encoding merged rule")
-			continue
-		}
-
-		if string(origData) != string(mergedData) {
-			logger.Debug().RawJSON("orig", origData).RawJSON("merged", mergedData).Msg("not the same, updating rule")
-			update = true
-		}
-
-		if update {
-			logger.Debug().Str("rule_id", rid).Msg("updating rule")
-			if err := makeRule(client, logger, ruleTemplate, false); err != nil {
-				logger.Warn().Err(err).Str("rule_id", rid).Interface("rule_cfg", ruleTemplate).Msg("updating rule")
-			}
-		} else {
-			logger.Debug().Str("rule_id", rid).Msg("unchanged, no update")
-		}
-	}
-
-	return nil
-}
-
-func createCustomRules(client *apiclient.API, logger zerolog.Logger, clusterName, clusterTag, checkCID string) error {
-	logger.Debug().Msg("create custom alerting rules")
-	configFile := viper.GetString(keys.CustomRulesFile)
-	data, err := ioutil.ReadFile(configFile)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return nil
-		}
-		logger.Warn().Err(err).Str("custom_rule_config", configFile).Msg("loading")
-		return err
-	}
-
-	var cr CustomRules
-	if err := json.Unmarshal(data, &cr); err != nil {
-		return err
-	}
-
-	create := true
-
-	// "managing" custom rules is more complex. for the moment, we'll rely on API to return
-	// an existing ruleset if the config matches.
-	for _, rule := range cr.Rules {
-		// create custom rules from configuration
-		if rule.CheckCID == "" {
-			rule.CheckCID = checkCID
-		}
-		rule.Name = strings.Replace(rule.Name, "{cluster_name}", clusterName, 1)
-		rule.Tags = append(rule.Tags, clusterTag)
-		if err := makeRule(client, logger, rule, create); err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-func makeRule(client *apiclient.API, logger zerolog.Logger, rule apiclient.RuleSet, create bool) error {
-
-	if create {
-		r, err := client.CreateRuleSet(&rule)
-		if err != nil {
-			return err
-		}
-		logger.Debug().Interface("rule", r).Msg("created rule set")
-	} else {
-		r, err := client.UpdateRuleSet(&rule)
-		if err != nil {
-			return err
-		}
-		logger.Debug().Interface("rule", r).Msg("updated rule set")
-	}
-
-	return nil
+const defaultRuleSetsStr117 = `
+{
```

Comment:
Kind of a nit, but there are inconsistent uses of tabs vs spaces in these constant strings.

Reply:
addressed in 8b8935a